README.md

```diff
@@ -1 +1,63 @@
-# interns-2grade
\ No newline at end of file
+# Board Game Training
+
+## Project Name
+**Board Game and Match Management System**
+
+## Endpoints
+
+### 1. Players - Route prefix: /players
+
+  - `GET /`: Retrieves information about all players.
+    - **Input**: N/A.
+    - **Output**: List of all PlayerDTO objects.
+    - **Exceptions**:
+      - Internal Server Error (Error 500): Internal database error.
+  - `GET /{nickname}`: Retrieves player information by nickname.
+    - **Input**: `nickname` parameter.
+    - **Output**: PlayerDTO object.
+    - **Exceptions**:
+      - Bad Request Exception (Error 400): The nickname field cannot be empty.
+      - Player not found (Error 404): No player is associated with that nickname.
+      - Internal Server Error (Error 500): Internal database error.
+  - `POST /`: Creates a new player.
+    - **Input**: JSON containing `name` and `nickname`.
+    - **Output**: PlayerDTO object.
+    - **Exceptions**:
+      - Bad Request Exception (Error 400): The name and nickname fields cannot be empty.
+      - Internal Server Error (Error 500): Internal database error.
+
+### 2. Games - Route prefix: /games
+
+  - `GET /`: Retrieves information about all games.
+    - **Input**: N/A.
+    - **Output**: List of all GameDTO objects.
+    - **Exceptions**:
+      - Internal Server Error (Error 500): Internal database error.
+  - `GET /{parameter}`: Retrieves game information by title, author, or age.
+    - **Input**: `parameter`.
+    - **Output**: List of GameDTO objects.
+    - **Exceptions**:
+      - Bad Request Exception (Error 400): The parameter field cannot be empty.
+      - Game not found (Error 404): No game matches the given parameters.
+      - Internal Server Error (Error 500): Internal database error.
+  - `POST /{name}`: Creates a game using the "name" parameter, fetching additional details from the public BGG API.
```

Comment:
For me, this doesn't make sense. We should use the same estructure for creating games that we are using for plays and players

I think that we should create a new endpoint to search in BGG, something like

```
GET /search?query={q}
```

The user could select one item in the list and then, we should add the game using the `POST /game` endpoint

Remember that all of games in the BGG has a important field called `bgg_id`. This is the way to check if the user has the game in the collection or not. You should send an error in when the game exists in the collection